src/goose.rs

```diff
@@ -713,43 +716,61 @@ impl GooseClient {
             weighted_on_start_tasks: Vec::new(),
             weighted_tasks: Vec::new(),
             weighted_on_stop_tasks: Vec::new(),
+            redirected_host: Arc::new(RwLock::new("".to_string())),
```

Comment:
I would not store the redirected_host, but the client base_url.

(Which can change over time and is initialized originally)

Reply:
This had much larger repercussions throughout the code, but the end effect is a net cleanup; it certainly makes sense to figure out the base_url one time per client instead of each time a URL is requested. Fixed.